lib/kafka/offset_manager.rb

```diff
@@ -111,17 +116,51 @@ def seconds_since_last_commit
       Time.now - @last_commit
     end
 
+    def seconds_since_last_recommit
+      Time.now - @last_recommit
+    end
+
     def committed_offset_for(topic, partition)
       @committed_offsets ||= @group.fetch_offsets
       @committed_offsets.offset_for(topic, partition)
     end
 
+    def offsets_to_commit(recommit = false)
+      if recommit && !@committed_offsets.nil?
```

Comment:
`#commit_offsets` sets `@committed_offsets` to nil, but `Consumer#fetch_batches` calls `OffsetManager#next_offset_for` for all subscribed partitions. So if there are any partitions that do not have `@processed_offsets` then `ConsumerGroup#fetch_offsets` will be called again to populate `@committed_offsets`.

`@resolved_offsets` are for partitions that are using the `:earliest` or `:latest` default and have to be resolved to an actual offset.